src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenStringConcat.cs

```diff
@@ -2104,5 +2105,1072 @@ .maxstack  5
   IL_0084:  ret
 }");
         }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/66827")]
+        public void ConcatTwo_ReadOnlySpan_LValue()
+        {
+            var source = """
+                using System;
+
+                public class Test
+                {
+                    static void Main()
+                    {
+                        var s = "s";
+                        var c = 'c';
+                        Console.Write(M1(s, c));
+                        Console.Write(M2(s, c));
+                    }
+
+                    static string M1(string s, char c) => s + c;
+                    static string M2(string s, char c) => c + s;
+                }
+                """;
+
+            var comp = CompileAndVerify(source, expectedOutput: RuntimeUtilities.IsCoreClr8OrHigherRuntime ? "sccs" : null, targetFramework: TargetFramework.Net80, verify: RuntimeUtilities.IsCoreClr8OrHigherRuntime ? default : Verification.Skipped);
```

Comment:
> We can still verify IL only in .NET 8

The word "still" is not correct here. We can verify IL on both .NET 8 and .NET Framework. We can't run .NET 8 program on .NET Framework host though, which is the reason why we skip output verification on .NET Framework. Yes, with this suggestion we will be able to verify output on both hosts, but in such case we would loose the ability to check IL on .NET Framework since it will fallback to `ToString` strategy due to the fact, that required members for the optimization are missing. This would be yet another test of the existing behavior before this PR, which is already covered by other tests in the file. So I think it is better to verify, that compiler, which runs on .NET Framework, correctly emits optimal IL with runtime references, containing required members for that.

> plus in those "missing member" tests, the IL should even be the same

Yes, it is true. However, currently tests verify, that missing at least 1 member, required for optimization, makes compiler take less optimal path. If we run .NET Framework tests with `NetStandard` references, we would loose this exact check on .NET Framework (since it doesn't have any of these members).